header.go

```diff
@@ -253,30 +254,48 @@ func whiteSpaceRune(r rune) bool {
 	return r == ' ' || r == '\t' || r == '\r' || r == '\n'
 }
 
-// Mirror of func in net/textproto/reader.go; from go 1.20.0-rc.2
+// validHeaderFieldByte Valid characters in email header field.
+//
+// According to [RFC 5322](https://www.rfc-editor.org/rfc/rfc5322#section-2.2),
+//
+//	> A field name MUST be composed of printable US-ASCII characters (i.e.,
+//	> characters that have values between 33 and 126, inclusive), except
+//	> colon.
 func validHeaderFieldByte(c byte) bool {
```

Comment:
should we eliminate this func and just call the new one you defined in internal/textproto?

Reply:
Will do. It was left there on purpose because we may need to remove `internal/textproto` someday if upstream fixes it.

Reply:
Yes, I had similar thought, but we probably will not switch back to upstream quickly.

Reply:
Would you like to save a patch file in enmime git repo to clearly indicate the changes compared to `net/textproto`? It might be easier for us to sync `internal/textproto` with upstream but still maintain our own version for email.

Reply:
Sounds good to me.

Reply:
@jhillyerd Could you help review again? also #284